gptask_cli/__init__.py

```diff
@@ -11,46 +11,58 @@ def check_file_staged_status(file, force):
         return False
     return True
 
-def get_files_to_process(recursive, file):
-    if file:
-        return [file]
-    if recursive:
-        return [open(f, 'r') for f in glob.glob(recursive, recursive=True)]
-    click.echo("Either a file or directory must be provided.")
-    return []
+def _get_path_list(path: str, is_recursive: bool):
+    """
+    Returns a list of paths from a glob pattern, file, or directory to recurse through.
+    """
+    if("*" in path):
+        return glob.glob(path, recursive=True)
+    elif os.path.isfile(path):
+        return [path]
+    elif(path[-1] == "/"):
+        path = path[:-1]
+    
+    # Recurse (or don't) through directory
+    return glob.glob(path + "/**/*" if is_recursive else path + "/*", recursive=True)
```

Comment:
ahhh yeah - this makes a lot more sense lmao

Reply:
Yeah I think it's cleaner for sure